pkg/master/master.go

```diff
@@ -78,13 +80,15 @@ const (
 
 // Config is a structure used to configure a Master.
 type Config struct {
-	EtcdHelper        tools.EtcdHelper
-	EventTTL          time.Duration
-	MinionRegexp      string
-	KubeletClient     client.KubeletClient
-	PortalNet         *net.IPNet
-	EnableLogsSupport bool
-	EnableUISupport   bool
+	EtcdHelper    tools.EtcdHelper
+	EventTTL      time.Duration
+	MinionRegexp  string
+	KubeletClient client.KubeletClient
+	PortalNet     *net.IPNet
+	// allow downstream consumers to disable the core controller loops
+	EnableCoreControllers bool
```

Comment:
Break the master up instead of making it this monolithic class?

----- Original Message -----
> > @@ -78,13 +80,15 @@ const (
> >  
> >  // Config is a structure used to configure a Master.
> >  type Config struct {
> > -	EtcdHelper        tools.EtcdHelper
> > -	EventTTL          time.Duration
> > -	MinionRegexp      string
> > -	KubeletClient     client.KubeletClient
> > -	PortalNet         *net.IPNet
> > -	EnableLogsSupport bool
> > -	EnableUISupport   bool
> > +	EtcdHelper    tools.EtcdHelper
> > +	EventTTL      time.Duration
> > +	MinionRegexp  string
> > +	KubeletClient client.KubeletClient
> > +	PortalNet     *net.IPNet
> > +	// allow downstream consumers to disable the core controller loops
> > +	EnableCoreControllers bool
> 
> I really dislike that we have 5 different patterns around this.  How do we
> unify this longer term.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7704/files#r29891511